src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer/Completion/RazorCompletionEndpoint.cs

```diff
@@ -470,6 +480,22 @@ internal static bool IsApplicableTriggerContext(CompletionContext context)
 
             completionItem = null;
             return false;
+
+            static bool IsInsertTextSnippet(RazorCompletionItem razorCompletionItem, out string insertText)
```

Comment:
Ahhh ya this is definnniitely a low cost way to add snippets into the completion experience. Not convinced it has to happen now but I think we should definitely consider bringing the understanding of snippets down into our completion provider system so that each of the completion providers can choose if they want to provide a snippet

Reply:
I agree that ideally this would live in the providers, but unless I'm missing something that's actually relatively challenging because the response here gets stored and re-constituted in like 3 different ways.

Reply:
Kind of, bringing it down into our provider level would mean expanding the [`RazorCompletionItem`](https://github.com/dotnet/razor-tooling/blob/340ee1728c3d0e4eca0a49681c113362c9e86ad7/src/Razor/src/Microsoft.CodeAnalysis.Razor.Workspaces/Completion/RazorCompletionItem.cs) type to have a `TextFormat` or something equivalent like property. Then at the re-building phase it'd do a direct translation from RazorTextFormat -> LSPTextFormat